### PR TITLE
Update contract and contracts-sdk getListingsForward

### PR DESCRIPTION
**Pull request by seansing9**
### Release Notes
[JIRA Link]()

#### Summary:
- Update MarketplaceListing class to include `slot` property
- Update contract's getListingsForward to include uint256 slot's data to capture uint256's.next slot value if there is only one listing for that tag
- Update contract's getListingsBackward to include 0 slot's data to capture 0.previous slot value if there is only one listing for that tag
- Update unit test to include a check that the listings front and back are returning incrementing / decrementing values correctly. 
- Clean up remnants of layer zero items and include erc7529 version to fix compiling issues within hardhat

### CODE REVIEW

File: packages/contracts-sdk/src/implementations/ConsentFactoryContract.ts [hunk 1]
```diff
@@ -365,22 +370,66 @@ export class ConsentFactoryContract
...
+          UnixTimestamp(Number(listings[i].timeExpiring)),
+          IpfsCID(cids[i]),
+          BigNumberString(listings[i].stake.toString()),
+          tag,
+        );
+
+        // First listing gets its slot from the second listing's previous value
+        if (i == 0) {
+          // Handle case where the startingSlot happens to be position before 0
+          if (listings[i + 1].previous == BigInt("0")) {
```
Comment (SnickerChar) [inline packages/contracts-sdk/src/implementations/ConsentFactoryContract.ts]: 0n is a bigint without the wrapper

File: packages/contracts/package.json [hunk 2]
```diff
@@ -15,19 +15,16 @@
...
   "type": "commonjs",
   "dependencies": {
-    "@layerzerolabs/lz-evm-oapp-v2": "^2.1.18",
-    "@layerzerolabs/lz-evm-protocol-v2": "^2.1.18",
-    "@layerzerolabs/solidity-examples": "^1.1.0",
     "@openzeppelin/contracts": "^5.0.2",
     "@openzeppelin/contracts-upgradeable": "^5.0.2",
     "@prb/math": "^4.0.2",
-    "@snickerdoodlelabs/erc7529": "workspace:^",
+    "@snickerdoodlelabs/erc7529": "2.0.1",
```
Comment (SnickerChar) [inline packages/contracts/package.json]: This should remain workspace:^

File: packages/contracts/test/README.md [hunk 3]
```diff
@@ -19,206 +19,3 @@ Only run tests in a given file:
 npx hardhat test test/consent.js
 
 ```
-
-# Testing OFT20Rewards
```
Comment (SnickerChar) [inline packages/contracts/test/README.md:23]: Why did you remove this?

File: packages/contracts/test/recommender.ts [hunk 4]
```diff
@@ -322,6 +322,31 @@ describe("Stake for Ranking tests", function () {
...
+      const listingsForward = await consentFactory.getListingsForward(
+        "NFT",
+        await token.getAddress(),
+        ethers.MaxUint256,
+        3,
+        true,
+      );
+
+      // Compare the next values for forward list
+      expect(listingsForward[1][0][1]).to.greaterThan(listingsForward[1][1][1]);
```
Comment (SnickerChar) [inline packages/contracts/test/recommender.ts]: Can you break out listingsForward[1][0][1] into some named values? const listings = listingsForward[1]; const Idunno = listings[0]; etc? Because as they are, these are completely opaque. This would be nice documentation of what is actually returned.

**Review by SnickerChar — DISMISSED** [review]
Approving as is, but had some minor comments.

File: packages/contracts/test/README.md (hunk 3 above)
Comment (seansing9) [reply packages/contracts/test/README.md:23]: Wanted to move this into our rewards package where all the Layer zero rewards sit, otherwise this part of the README doesn't work on this contracts package. Open to keeping a copy here as well though.

**Review by SnickerChar — APPROVED** [review]

**Review by orhunerd — APPROVED** [review]
